mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/entity/EntityHelperBase.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.entity;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.mapper.MapperContext;
+import com.datastax.oss.driver.api.mapper.MapperException;
+import com.datastax.oss.driver.api.mapper.annotations.DaoFactory;
+import com.datastax.oss.driver.api.mapper.annotations.DaoKeyspace;
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.api.mapper.entity.EntityHelper;
+
+public abstract class EntityHelperBase<EntityT> implements EntityHelper<EntityT> {
+
+  protected final CqlIdentifier keyspaceId;
+
+  protected final CqlIdentifier tableId;
+
+  protected final MapperContext context;
+
+  protected EntityHelperBase(MapperContext context, String tableName) {
+    this(context, null, tableName);
+  }
+
+  protected EntityHelperBase(MapperContext context, String keyspaceName, String tableName) {
+    this.context = context;
+    this.tableId =
+        context.getTableId() != null ? context.getTableId() : CqlIdentifier.fromCql(tableName);
+    if (keyspaceName == null) {
+      this.keyspaceId = context.getKeyspaceId();
+    } else {
+      this.keyspaceId =
+          context.getKeyspaceId() != null
+              ? context.getKeyspaceId()
+              : CqlIdentifier.fromCql(keyspaceName);
+    }
```

Comment:
I'm generally not a big fan of chained ternary operators, but in this case this might be a bit easier to follow:
```java
    this.keyspaceId =
        context.getKeyspaceId() != null
            ? context.getKeyspaceId()
            : (keyspaceName == null ? null : CqlIdentifier.fromCql(keyspaceName));
```